tests.md

```diff
@@ -65,6 +65,59 @@ end
 test_story = build(:story, :published_but_not_in_print)
 ```
 
+## Unit Vs Integration
+
+### Context
+
+We agree that unit tests will require one `it` definition for each expected behaviour.
+So that a `context` is allowed multiple scenarios.
+
+We agree that we will keep all expectations for a given request or integration test which touches the database context to a maximum of one scenario, with multiple expectations on it.
+This will allow us to test more scenarios (like the forms in the persistent object to being updated and the request status return) in one go and save test suite time.
+
+### Factories
+
+We have agreed that we will not use Factoris in our unit tests, this will make sure that no r/w to the DB is executed giving best performance.
+We can also make sure we control any signal that the code exchange with the integrated object.
+
+### Forms
+
+We have agreed that Forms functionality should be covered by pure unit tests as forms specs (`spec/forms`).
+We have agreed that Forms integrated functionality is tested through the `request`'s scenarios.
+
+### Requests
+
+We agree to use factories with traits in all our request specs.
+Our request specs will mainly check:
+  - Response status (redirect, success, failure)
+  - Persistence of the data after `POST`
+  - Errors raised by invalid parameters
+
+(? Share scenarios ?)
+
+### Services
```

Comment:
We agree we always unit test services.
We agree we will have integration tests for background jobs at a Task level

Reply:
Tags on integration tests and pop in the same file